sentry-android-core/src/main/java/io/sentry/android/core/internal/util/ScreenshotUtils.java

```diff
@@ -55,20 +61,57 @@ public class ScreenshotUtils {
       final Bitmap bitmap =
           Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.ARGB_8888);
 
-      final Canvas canvas = new Canvas(bitmap);
-      if (mainThreadChecker.isMainThread()) {
-        view.draw(canvas);
-      } else {
-        final @NotNull CountDownLatch latch = new CountDownLatch(1);
-        activity.runOnUiThread(
-            () -> {
-              try {
-                view.draw(canvas);
+      final @NotNull CountDownLatch latch = new CountDownLatch(1);
+
+      // Use Pixel Copy API on new devices, fallback to canvas rendering on older ones
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+
+        final HandlerThread thread = new HandlerThread("screenshot");
+        thread.start();
+
+        boolean success = false;
+        try {
+          final Handler handler = new Handler(thread.getLooper());
+          final AtomicBoolean copyResultSuccess = new AtomicBoolean(false);
+
+          PixelCopy.request(
+              window,
+              bitmap,
+              copyResult -> {
+                copyResultSuccess.set(copyResult == PixelCopy.SUCCESS);
                 latch.countDown();
-              } catch (Throwable e) {
-                logger.log(SentryLevel.ERROR, "Taking screenshot failed (view.draw).", e);
-              }
-            });
+              },
+              handler);
+
+          success =
+              latch.await(CAPTURE_TIMEOUT_MS, TimeUnit.MILLISECONDS) && copyResultSuccess.get();
+        } catch (InterruptedException e) {
```

Comment:
```suggestion
        } catch (Throwable e) {
```
I'd catch any throwable just to make sure

Reply:
I think that's fine as there's an outer try catch block already, which should take care of any other exceptions types